### PR TITLE
Add logging to job_runner.reporting

### LINKED ISSUE #1990: Replace network logger functionality for job runner
Replace network logging functionality in job runner / job_dispatch

### PR DESCRIPTION
**Pull request by sregales-TNO**
**Issue**
Resolves #1990

**Approach**
- Add a new logger to ert_logging/logger.conf
- Add logging to 
   - file module
   - interactive module
   - event module

### CODE REVIEW

File: ert_logging/logger.conf
Comment (oyvindeide) [inline ert_logging/logger.conf:29]: What is the main purpose of the new logging handle?

File: job_runner/reporting/interactive.py
Comment (oyvindeide) [inline job_runner/reporting/interactive.py:27]: This would work, but it relies on another reporter. I would suggest either writing a (very simple) custom reporter, or (I think) even better, add a `logger.info(msg)` here: https://github.com/equinor/ert/blob/8130abf0ae50dc4376c44429a04d22c5dba46609/job_runner/cli.py#L73

Then we leave it up to the logger to filter out data, and we dont impose restrictions in `ert`.

File: job_runner/reporting/file.py
Comment (oyvindeide) [inline job_runner/reporting/file.py:48]: It might be warranted, but do you think we need this many debug statements?

File: job_runner/reporting/event.py
Comment (oyvindeide) [inline job_runner/reporting/event.py:134]: I find this a bit harder to read than the previous version, although that is at the cost of some code duplication.

**Review by oyvindeide — CHANGES_REQUESTED** [review]
Could  you do a rebase on main?

File: ert_logging/logger.conf
Comment (sregales-TNO) [reply ert_logging/logger.conf:29]: I assume since job_runner module is a standalone top module, similar to ert_shared, it would need it's own logger. but if that is not the case it will be removed.

File: job_runner/reporting/event.py
Comment (sregales-TNO) [reply job_runner/reporting/event.py:134]: I was under the same impression when making the modification. But thought I put it in to see where the disscussion over optimilization vs readability will take me.

File: job_runner/reporting/file.py
Comment (sregales-TNO) [reply job_runner/reporting/file.py:48]: I was told at a standup that it is better to have too many and later with the team reduce to what is really needed

File: job_runner/reporting/interactive.py
Comment (sregales-TNO) [reply job_runner/reporting/interactive.py]: Not sure what is that needs to be done here. But I did the minimum you asked for and switch to info.

File: ert_logging/logger.conf
Comment (sregales-TNO) [reply ert_logging/logger.conf]: the logging handle is removed

**Comment by sregales-TNO** [comment]
Test this please

File: job_runner/reporting/event.py
Comment (oyvindeide) [reply job_runner/reporting/event.py:134]: So what is your thoughts on it now?

File: job_runner/reporting/interactive.py
Comment (oyvindeide) [reply job_runner/reporting/interactive.py]: Now the logging statement is inside the `Interactive` reporter, I would say it makes more sense to have it on line 74 of `ert/job_runner/cli.py` above, because then it is not tied to a specific reporter (as they serve a different purpose)

File: job_runner/reporting/event.py
Comment (jondequinor) [inline job_runner/reporting/event.py:77]: Why isn't there information about the exception here? It contains useful information.

File: job_runner/reporting/event.py
Comment (jondequinor) [inline job_runner/reporting/event.py:89]: This is ambiguous… Move it to `_dump_event` and change the line to

    logger.debug("schedule {type(event)} \"{event['type']}\" for delivery") 

or something?

File: job_runner/reporting/event.py
Comment (jondequinor) [inline job_runner/reporting/event.py:103]: I think all outcomes below will result in some debug statement, so this is maybe superfluous.

File: job_runner/reporting/event.py
Comment (jondequinor) [inline job_runner/reporting/event.py:118]: Is warning right here? The job failed, it's not a warning about anything. Either use `debug` or `error`.

https://dave.cheney.net/2015/11/05/lets-talk-about-logging

> Eliminate the warning level, it’s either an informational message, or an error condition.

File: job_runner/reporting/event.py
Comment (jondequinor) [inline job_runner/reporting/event.py:106]: But the `data` field varies. Sometimes there's JSON, sometimes it is `None`. From the CloudEvent spec:

> if the `datacontenttype` declares the data to contain JSON-formatted content, a JSON serializer MUST translate the data value to a JSON value, and use the member name data to store it inside the JSON representation. 

Even though omitting the `datacontettype` defaults to JSON if the format of the CloudEvent itself is JSON, I think it's best to omit `datacontettype` when `data` is `None`.

File: job_runner/reporting/event.py
Comment (jondequinor) [inline job_runner/reporting/event.py:127]: It has _exited_ this point.
```suggestion
            logger.debug(f"{job_name} job exited")
```

Maybe add exit code? Error message?

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:60]: It's not about the message, it's about the job… 
```suggestion
            logger.debug("Job {…} started")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:62]: Here, a `msg.success()` indicates that a job was successfully started.
```suggestion
                logger.debug("Job was successfully started")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:70]:
```suggestion
                logger.debug("Job {…} failed to start")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:80]:
```suggestion
```
Both outcomes (failed, success) of `Exited` is logged below.

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:83]:
```suggestion
                logger.debug("Job {…} exited successfully")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:87]:
```suggestion
                logger.debug("Job {…} failed with exit code {…} and error message {…}")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:102]:
```suggestion
            job_status.extend({
                "max_memory_usage": msg.max_memory_usage,
                "current_memory_usage": msg.current_memory_usage,
                "status": _JOB_STATUS_RUNNING,
            })
```
Maybe? Then you don't really need the inline function.

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:105]:
```suggestion
            logger.debug("Runner finished")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:107]:
```suggestion
                logger.debug("Runner finished successfully.")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:195]:
```suggestion
        logger.debug("Create an OK_file (All jobs complete)")
```

File: job_runner/reporting/file.py
Comment (jondequinor) [inline job_runner/reporting/file.py:166]: This is a bit implementation specific…

File: job_runner/reporting/interactive.py
Comment (jondequinor) [inline job_runner/reporting/interactive.py:27]: If the logging strategy should be this verbose, I think this line is a good approach. It logs what the reporter is doing without any interpretation. Maybe apply that to all logging for `job_runner`.

File: job_runner/reporting/event.py
Comment (sregales-TNO) [reply job_runner/reporting/event.py]: I think having the ternary operation within the function/method call is a bit too much.
But having multiple calls to the same function/method with similar arguments within a nested block can also be confusing, if not disoriented. So my solution was to reduce nested `if ... else` blocks if possible. With the latest change, I keep the ternary operation, but move it out of the method call and give distict labels that can be understood.

File: job_runner/reporting/interactive.py
Comment (sregales-TNO) [reply job_runner/reporting/interactive.py]: Adjustment to the code was made base on the following thread.
https://github.com/equinor/ert/pull/2028#discussion_r712747404

File: job_runner/reporting/event.py
Comment (sregales-TNO) [reply job_runner/reporting/event.py:107]: I added the following to `_dump_cloud_event`
```python
if data is None and _CONTENT_TYPE in attributes:
    attributes.pop(_CONTENT_TYPE)
```

File: job_runner/reporting/event.py
Comment (sregales-TNO) [reply job_runner/reporting/event.py:119]: agree

File: job_runner/reporting/file.py
Comment (sregales-TNO) [reply job_runner/reporting/file.py]: Inline function removed

File: job_runner/cli.py
Comment (oyvindeide) [inline job_runner/cli.py:76]: Can you move this outside the loop `reporters` loop? I probably told you to put it here.

File: job_runner/reporting/event.py
Comment (jondequinor) [reply job_runner/reporting/event.py:79]: It isn't caught, but it is logged (to disk).

File: job_runner/reporting/interactive.py
Comment (jondequinor) [inline job_runner/reporting/interactive.py:6]: What's the ignore for?

Comment (sregales-TNO) [reply job_runner/reporting/interactive.py:6]: There is no reason, it used to be when I was missing type hints. but I forgot to remove it

Comment (sregales-TNO) [reply job_runner/reporting/interactive.py:6]: ow wait, mistake, this is for pylint not mypy `job_runner/reporting/interactive.py:6:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)`

Comment (sregales-TNO) [reply job_runner/reporting/interactive.py:6]: it's because of the explicit 
```python
if ...:
  return
if ...:
  return
```
instead of a `if... else' block or `if ..: return ... return`

Comment (sregales-TNO) [reply job_runner/reporting/interactive.py:6]: So ignore will stay

Comment (sregales-TNO) [reply job_runner/reporting/interactive.py]: I removed it, but it fails on the none-strict version of mypy configuration

Comment (jondequinor) [reply job_runner/reporting/interactive.py:6]: What is the failure? I can't reproduce that.

Running `mypy -p ert -p ert3`

Comment (sregales-TNO) [reply job_runner/reporting/interactive.py:6]: That's interested unable to reproduce it locally, but it does fail on the pipeline.
Anyhow `job_runner/reporting/interactive.py:6: error: Missing return statement` is the error I will push it without it to show the problem

Comment (jondequinor) [reply job_runner/reporting/interactive.py]: You have outdated packages. Do `pip install --upgrade types-six typing-extensions` or something like it and you'll see the error.

You're missing a return statement for `_report`, because your checks on `isinstance(msg, type)` isn't exhaustive, so mypy expects to find some return at the end of the function.

Probably be easier to revert, or else you have to change the return signature to be `Optional[str]`, return `None` and then update `report` to either print or not…

**Review by jondequinor — APPROVED** [review]
LGTM! Thanks for your patience 😀

**Comment by jondequinor** [comment]
Be sure to rebase and squash those commits before merge.

**Comment by sregales-TNO** [comment]
Test this please

**Pull request merged**
